ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -728,27 +717,62 @@ public void shouldUseSerdeSupplierToBuildQueries() {
     );
   }
 
-  @Test
   @SuppressWarnings("unchecked")
-  public void shouldParseMultipleStatements() throws IOException {
-    final String statementsString = new String(Files.readAllBytes(
-        Paths.get("src/test/resources/SampleMultilineStatements.sql")), "UTF-8");
+  @Test
+  public void shouldHandleMultipleStatements() {
+    // Given:
+    final String sql = ""
+        + "-- single line comment\n"
+        + "/*\n"
+        + "   Multi-line comment\n"
+        + "*/\n"
+        + "CREATE STREAM S0 (a INT, b VARCHAR) "
+        + "      WITH (kafka_topic='s0_topic', value_format='DELIMITED');\n"
+        + "\n"
+        + "SET 'auto.offset.reset'='earliest';\n"
+        + "\n"
+        + "CREATE TABLE T1 (f0 BIGINT, f1 DOUBLE) "
+        + "     WITH (kafka_topic='t1_topic', value_format='JSON', key = 'f0');\n"
+        + "\n"
+        + "CREATE STREAM S1 AS SELECT * FROM S0;\n"
+        + "\n"
+        + "UNSET 'auto.offset.reset';\n"
+        + "\n"
+        + "CREATE STREAM S2 AS SELECT * FROM S0;\n"
+        + "\n"
+        + "DROP TABLE T1;";
+
+    givenTopicsExist("s0_topic", "t1_topic");
+
+    final List<QueryMetadata> queries = new ArrayList<>();
 
-    final List<Statement> statements =
-        ksqlEngine.parseStatements(statementsString)
-            .stream()
-            .map(PreparedStatement::getStatement)
-            .collect(Collectors.toList());
+    // When:
+    final List<PreparedStatement<?>> preparedStatements = ksqlEngine.parse(sql).stream()
+        .map(stmt ->
+        {
+          final PreparedStatement<?> prepared = ksqlEngine.prepare(stmt);
+          final ExecuteResult result = ksqlEngine.execute(prepared, KSQL_CONFIG, new HashMap<>());
+          result.getQuery().ifPresent(queries::add);
```

Comment:
I will willingly turn a blind eye to this 🙈

Reply:
lol